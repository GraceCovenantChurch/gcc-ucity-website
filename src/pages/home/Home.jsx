import React from "react";
import { useMediaQuery } from "react-responsive";
import Scroll from "react-scroll";

import { makeStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";

import ContentWithImage from "components/content/ContentWithImage";
import ContentWithImageAPI from "components/content/ContentWithImageAPI";
import CenteredContentWithGrid from "components/content/CenteredContentWithGrid";
import ConditionalAppURL from "components/mobileapp/ConditionalAppURL";
import MemoryVerseContent from "./content/MemoryVerseContent";
import EventsCard from "components/card/EventsCard";
import MobileEventsCard from "components/card/MobileEventsCard";
import ServicesCard from "components/card/ServicesCard";

import {
  getHomeEvents,
  massageEvents,
  // getCollegeFridayServices,
  // getSundayServices,
  // getCrossroadFridayServices,
  // massageServices,
  getReadingContent,
  massageReadingContent,
  getEasterServices,
  massageEasterServices,
} from "modules/Contentful";

import homeUrl from "static/images/home/worship.jpg";
import familyGroupUrl from "static/images/home/familygroup.jpg";
import amiQTUrl from "static/images/home/amiqt.jpg";
import { MOBILE_QUERY } from "constants/mobile";
import { HOME_DEFAULT } from "constants/home";

const Element = Scroll.Element;

const useStyles = makeStyles((theme) => ({
  eventsPadding: {
    padding: theme.spacing(2),
  },
}));

const Home = () => {
  const mobile = useMediaQuery({ query: MOBILE_QUERY });
  const classes = useStyles();
  const top = HOME_DEFAULT.top;
  const service = HOME_DEFAULT.service;
  const familygroup = HOME_DEFAULT.familygroup;
  const events = HOME_DEFAULT.events;
  const reading = HOME_DEFAULT.reading;

  return (
    <React.Fragment>
      <ContentWithImage
        isMobile={mobile}
        imageURL={homeUrl}
        title={top.vision}
        link={top.link}
      />
      <Element
        name={service.elementName}
        className={service.elementName}
        id={service.elementID}
      >
        {/* <CenteredContentWithGrid
          upperTitle
          isMobile={mobile}
          title={service.title}
          elementName={service.elementName}
          elementID={service.elementID}
          fetchCall={[
            getCollegeFridayServices,
            getSundayServices,
            getCrossroadFridayServices,
          ]}
          massage={massageServices}
          component={ServicesCard}
        /> */}
        <CenteredContentWithGrid
          upperTitle
          isMobile={mobile}
          title={"EASTER SERVICE LOCATION AND TIMES"}
          elementName={service.elementName}
          elementID={service.elementID}
          fetchCall={[getEasterServices]}
          massage={massageEasterServices}
          component={ServicesCard}
        />
      </Element>
      <ContentWithImage
        isMobile={mobile}
        imageURL={familyGroupUrl}
        title={familygroup.title}
        description={familygroup.description}
        link={familygroup.link}
      />
      <Container className={mobile ? classes.eventsMobilePadding : undefined}>
        <CenteredContentWithGrid
          upperTitle
          isMobile={mobile}
          title={events.title}
          fetchCall={[getHomeEvents]}
          massage={massageEvents}
          component={mobile ? MobileEventsCard : EventsCard}
          link={events.link}
        />
      </Container>
      <ContentWithImageAPI
        isMobile={mobile}
        title={reading.title}
        imageURL={amiQTUrl}
        fetchCall={[getReadingContent]}
        defaultDescription={ConditionalAppURL}
        massage={massageReadingContent}
        component={EventsCard}
        link={events.link}
      />
      <MemoryVerseContent isMobile={mobile} />
    </React.Fragment>
  );
};

export default Home;
