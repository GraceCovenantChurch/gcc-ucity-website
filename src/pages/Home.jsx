import React from "react";

import Page from "components/Page";

import { HOME_DEFAULT } from "constants/home";

const Home = () => {
  const top = HOME_DEFAULT.top;
  const service = HOME_DEFAULT.service;
  const familygroup = HOME_DEFAULT.familygroup;
  const events = HOME_DEFAULT.events;
  const reading = HOME_DEFAULT.reading;

  return (
    <Page>
      <div />
    </Page>
  );
};

export default Home;
