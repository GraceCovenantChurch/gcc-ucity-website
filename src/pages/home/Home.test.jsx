import React from "react";
import "__mocks__/matchMedia";
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import Home from "./Home";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({}),
  })
);

describe("Home", () => {
  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 300 }}>
          <Home />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(mobile).toMatchSnapshot();

    const { container: desktop } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 1000 }}>
          <Home />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(desktop).toMatchSnapshot();
  });
});
