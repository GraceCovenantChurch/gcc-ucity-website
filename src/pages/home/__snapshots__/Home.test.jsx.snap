// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div>
  <div
    class="background"
    style="background-image: url(test-file-stub);"
  >
    <div
      class="contentMobile"
    >
      <div
        class="MuiContainer-root makeStyles-mobileContentContainer-3 MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root makeStyles-title-7 makeStyles-mobileTitle-8 MuiTypography-body1"
        >
          Raising up kingdom workers
          <br />
          who are transformed by Christ
          <br />
          to influence the world
          <br />
        </p>
        <div
          class="makeStyles-linkContainer-4"
        >
          <a
            class="makeStyles-link-5 undefined"
            element="servicesContainer"
            scroll="true"
            text="Join us in worship >"
            url="services"
          >
            Join us in worship &gt;
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="services"
    id="servicesContainer"
    name="services"
  >
    <div
      class="MuiContainer-root makeStyles-container-12 MuiContainer-maxWidthLg"
    >
      <h5
        class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-h5"
      >
        SERVICE LOCATION AND TIMES
      </h5>
    </div>
  </div>
  <div
    class="background"
    style="background-image: url(test-file-stub);"
  >
    <div
      class="contentMobile"
    >
      <div
        class="MuiContainer-root makeStyles-mobileContentContainer-3 MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root makeStyles-title-7 makeStyles-mobileTitle-8 MuiTypography-body1"
        >
          Family Group
          <br />
        </p>
        <p
          class="MuiTypography-root makeStyles-description-9 makeStyles-mobileDescription-10 MuiTypography-body1"
        >
          You haven't checked out GCC unless
          <br />
          you've checked out our family groups.
          <br />
        </p>
        <div
          class="makeStyles-linkContainer-4"
        >
          <a
            class="makeStyles-link-5 undefined"
            scroll="false"
            text="Find out more >"
            url="/familygroup"
          >
            Find out more &gt;
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiContainer-root makeStyles-container-12 MuiContainer-maxWidthLg"
    >
      <h5
        class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-h5"
      >
        UPCOMING EVENTS
      </h5>
      <h5
        class="MuiTypography-root makeStyles-linkContainer-14 MuiTypography-h5"
      >
        <a
          class="makeStyles-learnMore-15"
          href="/events"
        >
          See all events &gt;
        </a>
      </h5>
    </div>
  </div>
  <div
    class="background"
    style="background-image: url(test-file-stub);"
  >
    <div
      class="contentMobile"
    >
      <div
        class="MuiContainer-root makeStyles-mobileContentContainer-17 MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root makeStyles-title-22 makeStyles-mobileTitle-23 MuiTypography-body1"
        >
          Read With Us
          <br />
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root makeStyles-container-26 MuiContainer-maxWidthLg"
  >
    <h5
      class="MuiTypography-root makeStyles-subtitle-27 MuiTypography-h5"
    >
      Monthly Memory Verse
    </h5>
  </div>
</div>
`;

exports[`Home matches the snapshot 2`] = `
<div>
  <div
    class="background"
    style="background-image: url(test-file-stub);"
  >
    <div
      class="contentMobile"
    >
      <div
        class="MuiContainer-root makeStyles-mobileContentContainer-3 MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root makeStyles-title-7 makeStyles-mobileTitle-8 MuiTypography-body1"
        >
          Raising up kingdom workers
          <br />
          who are transformed by Christ
          <br />
          to influence the world
          <br />
        </p>
        <div
          class="makeStyles-linkContainer-4"
        >
          <a
            class="makeStyles-link-5 undefined"
            element="servicesContainer"
            scroll="true"
            text="Join us in worship >"
            url="services"
          >
            Join us in worship &gt;
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="services"
    id="servicesContainer"
    name="services"
  >
    <div
      class="MuiContainer-root makeStyles-container-12 MuiContainer-maxWidthLg"
    >
      <h5
        class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-h5"
      >
        SERVICE LOCATION AND TIMES
      </h5>
    </div>
  </div>
  <div
    class="background"
    style="background-image: url(test-file-stub);"
  >
    <div
      class="contentMobile"
    >
      <div
        class="MuiContainer-root makeStyles-mobileContentContainer-3 MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root makeStyles-title-7 makeStyles-mobileTitle-8 MuiTypography-body1"
        >
          Family Group
          <br />
        </p>
        <p
          class="MuiTypography-root makeStyles-description-9 makeStyles-mobileDescription-10 MuiTypography-body1"
        >
          You haven't checked out GCC unless
          <br />
          you've checked out our family groups.
          <br />
        </p>
        <div
          class="makeStyles-linkContainer-4"
        >
          <a
            class="makeStyles-link-5 undefined"
            scroll="false"
            text="Find out more >"
            url="/familygroup"
          >
            Find out more &gt;
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiContainer-root makeStyles-container-12 MuiContainer-maxWidthLg"
    >
      <h5
        class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-h5"
      >
        UPCOMING EVENTS
      </h5>
      <h5
        class="MuiTypography-root makeStyles-linkContainer-14 MuiTypography-h5"
      >
        <a
          class="makeStyles-learnMore-15"
          href="/events"
        >
          See all events &gt;
        </a>
      </h5>
    </div>
  </div>
  <div
    class="background"
    style="background-image: url(test-file-stub);"
  >
    <div
      class="contentMobile"
    >
      <div
        class="MuiContainer-root makeStyles-mobileContentContainer-17 MuiContainer-maxWidthLg"
      >
        <p
          class="MuiTypography-root makeStyles-title-22 makeStyles-mobileTitle-23 MuiTypography-body1"
        >
          Read With Us
          <br />
        </p>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root makeStyles-container-26 MuiContainer-maxWidthLg"
  >
    <h5
      class="MuiTypography-root makeStyles-subtitle-27 MuiTypography-h5"
    >
      Monthly Memory Verse
    </h5>
  </div>
</div>
`;
